cheap-ruler/cheap-ruler.d.ts

```diff
@@ -0,0 +1,210 @@
+// Type definitions for cheap-ruler 2.4.1
+// Project: https://github.com/mapbox/cheap-ruler
+// Definitions by: Denis Carriere <https://github.com/DenisCarriere>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module "cheap-ruler" {
+    interface TemplateUnits {
+        kilometers: number
+        miles: number
+        nauticalmiles: number
+        meters: number
+        metres: number
+        yards: number
+        feet: number
+        inches: number
+    }
+    interface InterfacePointOnLine {
+        point: Array<number>
+        index: number
+        t: number
+    }
+    class CheapRuler {
+        /**
+         * Given two points of the form [longitude, latitude], returns the distance.
+         *
+         * @param {Array<number>} a point [longitude, latitude]
+         * @param {Array<number>} b point [longitude, latitude]
+         * @returns {number} distance
+         * @example
+         * var distance = ruler.distance([30.5, 50.5], [30.51, 50.49]);
+         * //=distance
+         */
+        distance(a: Array<number>, b: Array<number>): number;
```

Comment:
from the [code](https://github.com/mapbox/cheap-ruler/blob/master/index.js) it seems that `a` and `b` can be more precisely defined as tuples (`[number, number]`). the same is applicable for `bearing`, `destination` and few other methods below. For simplicity you can define type alias for the point type `type Point = [number, number]` and then use it everywhere where array mean point